src/main/java/com/google/api/generator/engine/ast/JavaDocComment.java

```diff
@@ -130,12 +136,16 @@ public boolean emptyComments() {
           && Strings.isNullOrEmpty(throwsDescription)
           && Strings.isNullOrEmpty(deprecated)
           && paramsList.isEmpty()
+          && Strings.isNullOrEmpty(returnText)
           && componentsList.isEmpty();
     }
 
     public JavaDocComment build() {
-      // @param, @throws and @deprecated should always get printed at the end.
+      // @param, @return, @throws and @deprecated should always get printed at the end.
       componentsList.addAll(paramsList);
+      if (!Strings.isNullOrEmpty(returnText)) {
+        componentsList.add(String.format("@return %s", returnText));
```

Comment:
This is great, and will be a big win for for the main branch. Any chance we could get this implemented in main, then merge the update from main into autoconfig-gen-draft2? (It may not be that simple, though -- since it looks like this branch still has to sync with the multimodule update.)

Reply:
Shouldn't this cause a large scale change in the golden files, or am I misunderstanding the scope of this change?

Reply:
@burkedavison with the multimodule update, it's not too feasible to merge `main` into `autoconfig-gen-draft2` (which will always stay as a separate branch) - but copying the relevant changes in a separate PR to `main` might be the easier way to go here? 

(For context: have a similar commit in this situation as part of #1208 - given that spring codegen plans to eventually move to use the gapic-generator-java jar, these changes do also need to be made in `main`; we are adding them directly to the spring branch for now given the current repo structure divergence.)

Reply:
This change is enabling us to add `@return` to the JavaDocs. You would still need to set `returnText` in the composers class for each method to actually generate the JavaDocs.
@burkedavison We decided that the SpringCodeGen will continue to work on a branch until preview release to prevent disruptions. That being said, I agree this change is important to the main branch as well and we should backport it, @zhumin8 can you please add it to the TODO list? We can create a PR against main with just this change.

Reply:
> Shouldn't this cause a large scale change in the golden files, or am I misunderstanding the scope of this change?

I think this change just enables setting return description in javadoc comments (a change that can be used by any of the gapic class comment composers, but the actual added calls to `setReturn` are scoped to the spring-specific ones).

Reply:
Thanks for clarifying!